src/lib/structures/expr.ml

```diff
@@ -1438,10 +1441,14 @@ and mk_forall_bis (q : quantified) =
         let q = {q with binders; user_trs = trs; sko_v; main = f } in
         mk_forall_bis q
 
+(* If [f] is a formula of the form [x = a -> P(x)] where [a] doesn't content
+   [x], this function produces the substitution { x |-> a }.
+
+   *Note*: this heuristic isn't used if the user has defined filters.
```

Comment:
Nit: the way to write bold text in odoc format is: `{b Note}`, I don't think `*Note*` works. (Doesn't matter here because this is not a doc comment, but odoc syntax is used below with `@return` so might as well be consistent)